### ISSUE TITLE
entity-event bug?

### ISSUE THREAD

**Issue by LisooYT** [SkriptLang/Skript#5581]
### Skript/Server Version

```markdown
[Skript] Skript's aliases can be found here: https://github.com/SkriptLang/skript-aliases
[Skript] Skript's documentation can be found here: https://docs.skriptlang.org/
[Skript] Skript's tutorials can be found here: https://docs.skriptlang.org/tutorials [Skript] Server Version: git-Paper-794 (MC: 1.16.5)
[Skript] Skript Version: 2.6.4
[Skript] Installed Skript Addons: None
[Skript] Installed dependencies: 
[Skript]  - WorldGuard v7.0.5+3827266
```

### Bug Description
```v
on spawn of a minecart with tnt:
    wait 5 seconds
    kill event-entity
```

and this doesnt work same with boats and other things i got some help from other ppl but still didnt work...
i guess vehicles dont work as event-entity and minecart tnt is an entity

### Expected Behavior

when placed/spawned minecart tnt
it should dissapear after 5 seconds

### Steps to Reproduce

i just writed that skript and it should work beacuse no errors were showing and other things

### Agreement

- [X] I have read the guidelines above and affirm I am following them with this report.

**Comment by AyhamAl-Ali** [comment]
I think this can be improved/solved by this PR #4329

**Comment by AyhamAl-Ali** [comment]
Does `delete event-entity` delete the entity?
